lib/services/notifications.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+const _ = require('lodash'),
+  log = require('../log'),
+  restler = require('restler'),
+  contentType = 'Content-type';
+
+/**
+ * @param {string} event
+ * @param {string} url
+ * @param {object} data
+ */
+function callWebhook(event, url, data) {
+  const headers = {
+      'X-Event': event
+    },
+    options = {
+      method: 'POST',
+      headers: headers
+    };
+
+  if (_.isPlainObject(data)) {
+    headers[contentType] = 'application/json';
+    options.body = JSON.stringify(data);
+  } else if (_.isString(data)) {
+    headers[contentType] = 'text/plain';
+    options.body = data;
+  }
+
+  console.log('callWebhook', event, url, data);
+
+  restler.request(url, options).on('complete', function (result, response) {
+    log.info('called webhook', _.pick(response, ['status', 'statusText']), result);
```

Comment:
Restler can be stubbed more easily.

Also, node-fetch is more generic and returns promises already.